### ISSUE TITLE
Support for HTML links

### ISSUE THREAD

**Issue by jimhester** [wilkelab/ggtext#16]
It would be great if we could use `<a href>` or markdown links and they would be included as links when saving plots using SVG output. There are some [existing solutions](https://stackoverflow.com/questions/42259826/hyperlinking-text-in-a-ggplot2-visualization) to this issue by modifying the produced SVG, but it seems more natural to do it with ggtext / gridtext.

``` r
library(ggplot2)
library(ggtext)
labels <- c(
  setosa = "<a href='https://en.wikipedia.org/wiki/Iris_setosa'>*I. setosa*</a>",
  virginica = "<a href='https://en.wikipedia.org/wiki/Iris_virginica'>*I. virginica*</a>",
  setosa = "<a href='https://en.wikipedia.org/wiki/Iris_versicolor'>*I. versicolor*</a>"
)

ggplot(iris, aes(Species, Sepal.Width)) +
  geom_boxplot() +
  scale_x_discrete(
    name = NULL,
    labels = labels
  ) +
  theme(
    axis.text.x = element_markdown(color = "black", size = 11)
  )
#> Error: gridtext has encountered a tag that isn't supported yet: <a>
#> Only a very limited number of tags are currently supported.
```

<sup>Created on 2020-02-06 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by clauswilke** [comment]
I agree. The problem is actually the graphics device. I don't think it's currently possible to push a link through an R graphics device.

The fundamental problem is that all I have at my disposal is elementary grid grobs, `textGrob()`, `rectGrob()`, etc. If something can't be done with those grobs I can't do it in gridtext.

**Comment by jimhester** [comment]
Yeah, I figured that might be an issue, makes sense.

**Comment by rensa** [comment]
Not sure if this is helpful, but the `gridSVG` package includes a function, `grid.garnish`, that lets you add SVG attributes to nodes on the grid tree. [I wrote about using it to inject semantics (ie. the original data)](https://medium.com/@rensa/challenges-visualising-climate-extremes-with-leaflet-7a407162d75f) into SVG elements for further visualisation.

I never generalised this technique because different geoms produce different numbers of elements (and even then one geom can vary depending on the data it's marking), and this use case would be different because you'd need additional SVG _elements_ (ie. `<a>` elements), not just attributes on existing elements. But maybe there's a way there?
